packages/base-controller/src/BaseController.ts

```diff
@@ -120,17 +120,17 @@ export class BaseController<C extends BaseConfig, S extends BaseState> {
         ? (config as C)
         : Object.assign(this.internalConfig, config);
 
-      for (const key in this.internalConfig) {
+      for (const key of Object.keys(this.internalConfig) as (keyof C)[]) {
         if (typeof this.internalConfig[key] !== 'undefined') {
-          (this as any)[key as string] = this.internalConfig[key];
+          (this as any)[key] = this.internalConfig[key];
         }
       }
     } else {
-      for (const key in config) {
+      for (const key of Object.keys(config) as (keyof C)[]) {
```

Comment:
This one is a weird one and I couldn't get it to be any better than this :(

Reply:
I think `Object.entries()` doesn't work if the key needs to be more specific than `string`.

Reply:
Yeah, this one is different too I think because we're using the keys from one object to operate on another object and both objects have specific types for their keys.